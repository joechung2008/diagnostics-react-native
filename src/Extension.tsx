import React from 'react';
import { ScrollView, Text } from 'react-native';
import Configuration from './Configuration';
import StageDefinition from './StageDefinition';
import { useTheme } from './ThemeContext';

const Extension: React.FC<ExtensionProps> = ({
  config,
  extensionName,
  stageDefinition,
}) => {
  const { colors } = useTheme();

  const dynamicStyles = {
    scrollView: {
      flex: 1,
      backgroundColor: colors.background,
    },
    container: {
      display: 'flex' as const,
      flexDirection: 'column' as const,
      gap: 16,
      padding: 10,
    },
    title: {
      fontSize: 18,
      fontWeight: '600' as const,
      marginBottom: 10,
      color: colors.text,
    },
  };

  return (
    <ScrollView
      style={dynamicStyles.scrollView}
      contentContainerStyle={dynamicStyles.container}
    >
      <Text style={dynamicStyles.title}>{extensionName}</Text>
      {config && <Configuration config={config} />}
      {stageDefinition && <StageDefinition stageDefinition={stageDefinition} />}
    </ScrollView>
  );
};

export default Extension;
