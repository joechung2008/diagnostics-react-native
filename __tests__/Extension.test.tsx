import Extension from '@/Extension';
import { render, screen } from '@testing-library/react-native';
import React from 'react';

// Mock the ThemeContext
jest.mock('@/ThemeContext', () => ({
  useTheme: () => ({
    colors: {
      background: '#ffffff',
      text: '#212529',
    },
  }),
}));

// Mock child components
jest.mock('@/Configuration', () => 'Configuration');
jest.mock('@/StageDefinition', () => 'StageDefinition');

describe('Extension', () => {
  it('renders correctly with all props', () => {
    const props = {
      extensionName: 'test-extension',
      manageSdpEnabled: true,
      config: {
        apiUrl: 'https://api.example.com',
        timeout: '30000',
      },
      stageDefinition: {
        development: ['dev1', 'dev2'],
        production: ['prod1'],
      },
    };
    render(<Extension {...props} />);
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('renders correctly with minimal props', () => {
    const props = {
      extensionName: 'minimal-extension',
      manageSdpEnabled: false,
    };
    render(<Extension {...props} />);
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('renders correctly with config only', () => {
    const props = {
      extensionName: 'config-only-extension',
      manageSdpEnabled: true,
      config: {
        debug: 'true',
        logLevel: 'info',
      },
    };
    render(<Extension {...props} />);
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('renders correctly with stage definition only', () => {
    const props = {
      extensionName: 'stage-only-extension',
      manageSdpEnabled: true,
      stageDefinition: {
        staging: ['stage1'],
        production: ['prod1', 'prod2'],
      },
    };
    render(<Extension {...props} />);
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
