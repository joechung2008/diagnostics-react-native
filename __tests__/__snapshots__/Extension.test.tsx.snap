// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Extension renders correctly with all props 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "display": "flex",
      "flexDirection": "column",
      "gap": 16,
      "padding": 10,
    }
  }
  style={
    {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <View>
    <Text
      style={
        {
          "color": "#212529",
          "fontSize": 18,
          "fontWeight": "600",
          "marginBottom": 10,
        }
      }
    >
      test-extension
    </Text>
    <Configuration
      config={
        {
          "apiUrl": "https://api.example.com",
          "timeout": "30000",
        }
      }
    />
    <StageDefinition
      stageDefinition={
        {
          "development": [
            "dev1",
            "dev2",
          ],
          "production": [
            "prod1",
          ],
        }
      }
    />
  </View>
</RCTScrollView>
`;

exports[`Extension renders correctly with config only 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "display": "flex",
      "flexDirection": "column",
      "gap": 16,
      "padding": 10,
    }
  }
  style={
    {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <View>
    <Text
      style={
        {
          "color": "#212529",
          "fontSize": 18,
          "fontWeight": "600",
          "marginBottom": 10,
        }
      }
    >
      config-only-extension
    </Text>
    <Configuration
      config={
        {
          "debug": "true",
          "logLevel": "info",
        }
      }
    />
  </View>
</RCTScrollView>
`;

exports[`Extension renders correctly with minimal props 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "display": "flex",
      "flexDirection": "column",
      "gap": 16,
      "padding": 10,
    }
  }
  style={
    {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <View>
    <Text
      style={
        {
          "color": "#212529",
          "fontSize": 18,
          "fontWeight": "600",
          "marginBottom": 10,
        }
      }
    >
      minimal-extension
    </Text>
  </View>
</RCTScrollView>
`;

exports[`Extension renders correctly with stage definition only 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "display": "flex",
      "flexDirection": "column",
      "gap": 16,
      "padding": 10,
    }
  }
  style={
    {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <View>
    <Text
      style={
        {
          "color": "#212529",
          "fontSize": 18,
          "fontWeight": "600",
          "marginBottom": 10,
        }
      }
    >
      stage-only-extension
    </Text>
    <StageDefinition
      stageDefinition={
        {
          "production": [
            "prod1",
            "prod2",
          ],
          "staging": [
            "stage1",
          ],
        }
      }
    />
  </View>
</RCTScrollView>
`;
